Source/core/JSON.h

```diff
@@ -4980,6 +4979,114 @@ namespace Core {
+                JSON::String::operator=(std::move(move));
+                return (*this);
+            }
+
+            ThreadId& operator=(const ThreadId& RHS)
+            {
+                JSON::String::operator=(RHS);
+                return (*this);
+            }
+
+            template <typename TYPE>
+            inline typename std::enable_if<(std::is_same<TYPE, void*>::value), ThreadId&>::type
+            operator=(const TYPE& RHS)
+            {
+                String::operator=(Core::NumberType<uintptr_t, false, NumberBase::BASE_HEXADECIMAL>(reinterpret_cast<uintptr_t>(RHS)).Text());
+                return (*this);
+            }
+
+            template <typename TYPE>
+            inline typename std::enable_if<(!std::is_same<TYPE, void*>::value), ThreadId&>::type
+            operator=(const TYPE RHS)
+            {
+                String::operator=(Core::NumberType<TYPE, false, NumberBase::BASE_HEXADECIMAL>(RHS).Text());
+                return (*this);
+            }
+
+            template <typename TYPE>
+            inline typename std::enable_if<(std::is_same<TYPE, void*>::value), thread_id>::type
+            Default() const
+            {
+                return reinterpret_cast<thread_id>(Core::NumberType<uintptr_t, false, NumberBase::BASE_HEXADECIMAL>(String::Default().c_str(), static_cast<uint32_t>(String::Default().length())).Value());
+            }
+
+            template <typename TYPE>
+            inline typename std::enable_if<!(std::is_same<TYPE, void*>::value), thread_id>::type
+            Default() const
+            {
+                return static_cast<thread_id>(Core::NumberType<uint64_t, false, NumberBase::BASE_HEXADECIMAL>(String::Default().c_str(), static_cast<uint32_t>(String::Default().length())).Value());
+            }
+
+            inline thread_id Default() const
+            {
+                return Default<thread_id>();
+            }
+
+            template <typename TYPE>
+            inline typename std::enable_if<(std::is_same<TYPE, void*>::value), thread_id>::type
+            Value() const
+            {
+                return reinterpret_cast<thread_id>(Core::NumberType<uintptr_t, false, NumberBase::BASE_HEXADECIMAL>(String::Value().c_str(), static_cast<uint32_t>(String::Value().length())).Value());
+            }
+
+            template <typename TYPE>
+            inline typename std::enable_if<!(std::is_same<TYPE, void*>::value), thread_id>::type
+            Value() const
+            {
+                return static_cast<thread_id>(Core::NumberType<uint64_t, false, NumberBase::BASE_HEXADECIMAL>(String::Value().c_str(), static_cast<uint32_t>(String::Value().length())).Value());
+            }
+
+            inline thread_id Value() const
+            {
+                return Value<thread_id>();
+            }
+
+            inline operator thread_id() const
+            {
+                return Value();
+            }
+        };
+
+        typedef ThreadId Pointer;
```

Comment:
Perhaps it's better to remove the Pointer typedef all along? As it's either gonna be InstanceId or ThreadId and update the code that's using it accordingly.